pkg/controllers/vdb/upgrade.go

```diff
@@ -680,33 +680,81 @@ func (i *UpgradeManager) closeAllSessions(ctx context.Context, pfacts *PodFacts)
+	"or all_roles like '%%, dbadmin, %%' " +
+	"or all_roles like 'dbadmin*' " +
+	"or all_roles like '%%, dbadmin*' " +
+	"or all_roles like 'dbadmin*, %%' " +
+	"or all_roles like '%%, dbadmin*, %%')"
+
 // areAllConnectionsPaused will run a query to see the number of non-superuser connections that are active (not paused)
 // it returns a requeue error if there are still active connections
-func (i *UpgradeManager) areAllConnectionsPaused(ctx context.Context, pfacts *PodFacts) (ctrl.Result, error) {
+func (i *UpgradeManager) areAllConnectionsPaused(ctx context.Context, pfacts *PodFacts) (bool, error) {
 	pf, ok := pfacts.findFirstUpPod(true, "")
 	if !ok {
 		i.Log.Info("No pod found to run vsql. Waiting for cluster to come up")
-		return ctrl.Result{}, nil
+		return false, nil
 	}
 
 	sql := "select sum(c.active - s.sessions) from " +
 		"(select node_name, user_sessions - paused_sessions as active from v_internal.vs_client_connections) c " +
 		"join (select node_name, count(*) as sessions from v_monitor.sessions join v_catalog.users using (user_name) " +
-		"where all_roles like '%%dbadmin%%' group by node_name) s " +
+		"where " + userHasDbadminRole + " group by node_name) s " +
 		"using (node_name)"
 
 	cmd := []string{"-tAc", sql}
 	stdout, _, err := pfacts.PRunner.ExecVSQL(ctx, pf.name, names.ServerContainer, cmd...)
 	if err != nil {
-		return ctrl.Result{}, err
+		return false, err
 	}
 
 	active := anyActiveConnections(stdout)
 	if active {
 		i.Rec.Eventf(i.Vdb, corev1.EventTypeWarning, events.PauseConnectionsRetry,
 			"Cluster still has active connections. Retrying pause connections.")
 	}
-	return ctrl.Result{Requeue: active}, nil
+	return active, nil
+}
+
+// closeAllUnpausedSessions will run a query to close all active non-dbadmin sessions.
+func (i *UpgradeManager) closeAllUnpausedSessions(ctx context.Context, pfacts *PodFacts) error {
+	pf, ok := pfacts.findFirstPodSorted(func(v *PodFact) bool {
+		return v.isPrimary && v.upNode
+	})
+	if !ok {
+		i.Log.Info("No pod found to run vsql. Skipping close all sessions")
+		return nil
+	}
+
+	sql := "select s.session_id from " +
+		"v_internal.vs_sessions s join v_catalog.users u using (user_name) join v_monitor.transactions t using (transaction_id) " +
+		"where not s.is_paused and not " + userHasDbadminRole
+	sessionIds, stderr, err := pfacts.PRunner.ExecVSQL(ctx, pf.name, names.ServerContainer, "-tAc", sql)
+	if err != nil {
+		i.Log.Error(err, "failed to retrieve unpaused sessions", "stderr", stderr)
+		return err
+	}
+
+	errs := []error{}
+	for _, id := range strings.Split(strings.TrimSuffix(sessionIds, "\n"), "\n") {
+		if id == "" {
+			continue
+		}
+		killCmd := []string{"-tAc", fmt.Sprintf("select close_session('%s')", id)}
+		_, stderr, err = pfacts.PRunner.ExecVSQL(ctx, pf.name, names.ServerContainer, killCmd...)
+		if err != nil {
+			i.Log.Error(err, "failed to kill session", "session_id", id, "stderr", stderr)
+			errs = append(errs, err)
```

Comment:
Define errs as an error, then use "errs = errors.Join(errs, err) to collect all errors.